core/worker.cc

```diff
@@ -291,7 +291,14 @@ void *Worker::Run(void *_arg) {
   wid_ = arg->wid;
   core_ = arg->core;
   socket_ = rte_socket_id();
-  CHECK_GE(socket_, 0);  // shouldn't be SOCKET_ID_ANY (-1)
+
+  // For some reason, rte_socket_id() does not return a correct NUMA ID.
```

Comment:
This is a bit troubling, I can't see why `rte_socket_id` should fail here, but the change itself seems OK.

Reply:
Oh my bad. "does not" should have been "may not"

Reply:
I figured that - obviously this _usually_ works. Under what condition, I wonder, is it failing?